test/src/testcontents.ts

```diff
@@ -103,6 +102,64 @@ describe('jupyter.services - Contents', () => {
 
   });
 
+  describe('.getAbsolutePath()', () => {
+
+    it('should get a file in the base directory', () => {
+      let path = ContentsManager.getAbsolutePath('bar.txt');
+      expect(path).to.be('bar.txt');
+    });
+
+    it('should get a file in the current directory', () => {
+      let path = ContentsManager.getAbsolutePath('./bar.txt', 'baz');
+      expect(path).to.be('baz/bar.txt');
+    });
+
+    it('should get a file in the parent directory', () => {
+      let path = ContentsManager.getAbsolutePath('../bar.txt', '/fizz/buzz');
+      expect(path).to.be('fizz/bar.txt');
+    });
+
+    it('should get a file in the grandparent directory', () => {
+      let path = ContentsManager.getAbsolutePath('../../bar.txt', 'fizz/buzz/bing/');
+      expect(path).to.be('fizz/bar.txt');
+    });
+
+    it('should return `null` if not contained in the base url', () => {
+      let path = ContentsManager.getAbsolutePath('../../bar.txt', 'fizz');
+      expect(path).to.be(null);
+    });
+
+    it('should short-circuit to the root directory of the server', () => {
+      let path = ContentsManager.getAbsolutePath('/bar.txt', 'fizz/buzz');
+      expect(path).to.be('bar.txt');
+    });
+
+    it('should yield the current directory', () => {
+      let path = ContentsManager.getAbsolutePath('.', 'fizz/buzz');
+      expect(path).to.be('fizz/buzz');
+    });
+
+    it('should yield the parent directory', () => {
+      let path = ContentsManager.getAbsolutePath('..', 'fizz/buzz');
+      expect(path).to.be('fizz');
+    });
+
```

Comment:
Can we make a test showing that things are *not* URL encoded? Perhaps a path with a space and a question mark in it?